### ISSUE TITLE
find() no criteria and fields/select provided does not work

### ISSUE THREAD

**Issue by mahadazad** [balderdashy/waterline#1098]
I am trying to fetch all the records but with selected fields, I have tried the following ways but none works:

```
Post.find(
    {
       where: {},
       select: ['title']
    }
);
```
```
Post.find(
    {},
    {
      fields: {
          title: 1
       }
    }
);
```

**Comment by dmarcelino** [comment]
Are you using sails-mongo? From https://github.com/balderdashy/waterline/issues/1059#issuecomment-112361427:

> Sails-mongo did not support select until recently (balderdashy/sails-mongo#281), a fix has been made and it will go out in the next release (v0.11.3 I believe).

We probably should push a new sails-mongo version as this is the 2nd or 3rd issue about it...

**Comment by mahadazad** [comment]
I am using 0.11.2 the select works if I pass some criteria e.g. ```category: 'food'``` but it does not work if I pass empty criteria i.e. ```{}``` or ```null```

**Comment by dmarcelino** [comment]
can you try sails-mongo master branch: `npm i balderdashy/sails-mongo`?

**Comment by mahadazad** [comment]
Its working with the latest version.

Thanks

**Comment by mahadazad** [comment]
Ok found the issue. It only works if you also give it some criteria or limit or order

e.g.

this wont work:
```
Post.find(
    {},
    {fields: {title: 1}}
)
```

but these will:
```
Post.find(
    {categories: 'food'},
    {fields: {title: 1}}
)
```

```
Post.find(
    {},
    {fields: {title: 1}}
)
.limit(10)
```


```
Post.find(
    {},
    {fields: {title: 1}}
)
.order({date: -1})
```

**Comment by devinivy** [comment]
Sounds like a great candidate for a failing test in https://github.com/balderdashy/waterline-adapter-tests!  I imagine this is a bug in sails-mongo rather than waterline, but I can't verify that at the moment.

**Comment by dmarcelino** [comment]
@mahadazad, I'm a bit confused with your syntax... where in Waterline did you see `fields` keyword being used as you suggest in:
```js
Post.find(
    {},
    {fields: {title: 1}}
)
```

The correct syntax, as per the [waterline docs - select](https://github.com/balderdashy/waterline-docs/blob/master/queries/query-language.md#select) and the [select tests](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/queryable/modifiers/select.modifier.test.js#L39) is:
```js
Post.find(
    {},
    select: ['title']
)
```

Can you try that and report back? Thanks

**Comment by mahadazad** [comment]
Actually 'fields' work 'select' does not work. May be its due to sails-mongo?

**Comment by dmarcelino** [comment]
> Actually 'fields' work 'select' does not work. May be its due to sails-mongo?

@mahadazad, were you using the master branch version (`npm i balderdashy/sails-mongo`)?

I've just created an additional test balderdashy/waterline-adapter-tests/pull/84 that performs the query `Queryable.User.find({ where: {}, select: ['first_name'], sort: 'age' }, function(err, users) {/*...*/}` and mongodb (master version) passed this test: https://travis-ci.org/balderdashy/waterline-adapter-tests/jobs/72499276#L564.

**Comment by dmarcelino** [comment]
Merged PR balderdashy/waterline-adapter-tests#84 and closing this for now, please re-open if you feel this hasn't been fully answered.
